src/app/shared/reuseHelper.js

```diff
@@ -50,8 +54,23 @@ module.exports = {
     },
   },
   generateUserDetails: (userDetailsResponse, i18n) => {
+    let givenName = "";
+    let familyName = "";
+    if (userDetailsResponse.data?.nameParts) {
+      userDetailsResponse.data.nameParts.forEach((namePart) => {
+        if (namePart.type === "GivenName") {
+          givenName = namePart.value;
+        } else if (namePart.type === "FamilyName") {
+          familyName = namePart.value;
+        }
+      });
```

Comment:
I don't think this is right - what if you have multiple given names?

What about something like:
```
const givenName = userDetailsResponse.data?.nameParts
  .filter((namePart) => namePart.type === "GivenName")
  .join(" ");
```

Reply:
Went with a slightly different approach, idea is to try reducing the operations. Instead of using two separate `filter`/`find` operations